### PR TITLE
Added Ring of Arcana

### PR DESCRIPTION
**Pull request by condorcraft110**
I hope I got all the features, it's a complex little ring x3n0 had going on there ;)

### CODE REVIEW

**Comment by sinkillerj** [comment]
@williewillus I'm not seeing any issues here, but give it a eye over and make sure it is up to your refactor standards.

**Comment by condorcraft110** [comment]
I also attempted to fix #721 by using my code from the Ring of Arcana.

File: src/main/java/moze_intel/projecte/gameObjs/entity/EntityFireProjectile.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	{
+		super(world);
+	}
+	
+	public EntityFireProjectile(World world, EntityLivingBase player)
+	{
+		super(world, player);
+	}
+	
+	protected void onImpact(MovingObjectPosition mop)
```
Comment (williewillus) [inline src/main/java/moze_intel/projecte/gameObjs/entity/EntityFireProjectile.java]: @Override annotation

File: src/main/java/moze_intel/projecte/gameObjs/entity/EntityLightningProjectile.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	}
+	
+	public EntityLightningProjectile(World world, EntityLivingBase player, ItemStack ring)
+	{
+		super(world, player);
+		
+		this.ring = ring;
+	}
+	
+	protected void onImpact(MovingObjectPosition mop)
```
Comment (williewillus) [inline src/main/java/moze_intel/projecte/gameObjs/entity/EntityLightningProjectile.java]: @Override annotation

File: src/main/java/moze_intel/projecte/gameObjs/items/rings/Arcana.java [hunk 3]
```diff
@@ -1,155 +1,263 @@
...
+	}
+	
+	public boolean doesContainerItemLeaveCraftingGrid(ItemStack stack)
+	{
+		return false;
 	}
 	
-	@Override
-	public void onUpdate(ItemStack stack, World world, Entity entity, int slotIndex, boolean isHeld)
+	public byte getMode(ItemStack stack)
```
Comment (williewillus) [inline src/main/java/moze_intel/projecte/gameObjs/items/rings/Arcana.java]: @Override annotation on all of these that override superclass metgods

**Comment by williewillus** [comment]
I got kazy commenting on every occurence of it, but every overridden method should have @Override on it. This is purely a safety check, but is a VERY good one to have. I.E. if Mojang changes an item method name, which they do - a lot - in 1.8, then without the annotation the items would recompile and then silently fail at everything during runtime. Having Override enforces our method's replacement of a super method and makes things easier and less-headache prone in the long run. Compilation would fail if mojang changed names or the mcp mapping shifts

**Comment by condorcraft110** [comment]
@williewillus done.

**Comment by sinkillerj** [comment]
@condorcraft110 Few merge conflicts to resolve then I can merge this.

**Comment by condorcraft110** [comment]
@sinkillerj I haven't used Github much - how do I check what files are conflicting?

**Comment by williewillus** [comment]
On your local working copy, pull in the latest origin/master and merge it into your branch. It should then tell you you have conflicts and leave an incomplete working copy with the conflicted files shown.

File: src/main/java/moze_intel/projecte/gameObjs/items/rings/Arcana.java [hunk 4]
```diff
@@ -1,155 +1,280 @@
...
+	public void addInformation(ItemStack stack, EntityPlayer player, List list, boolean b)
 	{
-		if (!world.isRemote)
+		if(stack.hasTagCompound())
 		{
-			if (stack.stackTagCompound.getBoolean("Active"))
+			if(!stack.stackTagCompound.getBoolean("Active"))
 			{
-				stack.stackTagCompound.setBoolean("Active", false);
+				list.add(EnumChatFormatting.RED + "Inactive!");
```
Comment (williewillus) [inline src/main/java/moze_intel/projecte/gameObjs/items/rings/Arcana.java]: I missed this the first time, but no raw English strings in the code. Do ``` + StatCollector.translateToLocal("pe.arcana.inactive");``` and then put ```pe.arcana.inactive=Inactive!``` in the en_US.lang. So it lets the string be translated to other languages

**Comment by condorcraft110** [comment]
I also apparently managed to commit every single file changed since I submitted the pullreq, and there are still merge conflicts. I'll try again, I guess.

**Comment by condorcraft110** [comment]
@sinkillerj I seem to have fixed the conflicts; have a look over it, ee if I messed anything up ;)

**Pull request merged**